xbmc/Application.h

```diff
@@ -125,7 +122,8 @@ class CApplication : public IWindowManagerCallback,
                      public ISettingCallback,
                      public ISettingsHandler,
                      public ISubSettings,
-                     public KODI::MESSAGING::IMessageTarget
+                     public KODI::MESSAGING::IMessageTarget,
+                     public CApplicationPowerHandling
```

Comment:
I assume you're using inheritance for composition? Sometimes there's better ways to compose components.

Reply:
yes. but i don't want to change the approach now. i just want to split the class in logical bits, then we can see about is-a vs has-a later.